app/models/services/locators/doi_paper_locator.rb

```diff
@@ -6,30 +6,32 @@ def column
   def find_paper
     return super if super
 
+    paper = Paper.new
+
+    crossref = Crossref.new
+    if result = crossref.get_doi_metadata(self.locator_id)
+      data = result['message']
+      paper = self.import_data_to_paper(paper,data,'crossref')
+    end
+
     pubmed = Pubmed.new
     if uid = pubmed.find_uid_by_doi(self.locator_id)
       result = pubmed.get_uid_metadata(uid)
       data = result['result'][uid]
-      names = data['authors'].map {|a| a['name']}
+      paper = self.import_data_to_paper(paper,data,'pubmed')
+    end
 
-      existing_authors = Author.where(name: names)
-      existing_names = existing_authors.map(&:name)
-      new_authors = (names - existing_names).map {|a| Author.create name: a}
+    existing_paper = Paper.where( "title = ? or pubmed_id = ? and pubmed_id is not null", paper.title, paper.pubmed_id )
```

Comment:
I would prefer the Rails finders here. `Paper.where(title: paper.title, pubmed_id: paper.pubmed_id).not(pubmed_id: nil)`, although something about that strikes me as odd, anyway. If you're using the paper's pubmed id, why do you have to check if the pubmed id is null?